e2e/tools/validator/src/validator/cli/__init__.py

```diff
@@ -610,6 +611,34 @@ def validate_acpi(cfg: config.Validator, duration: datetime.timedelta, report_di
     return int(res.validations.passed)
 
 
+@validator.command()
+@click.option("--duration", "-d", type=options.Duration(), required=True)
+# ruff: noqa: S108 (Suppressed as we are intentionally using `/tmp` as reporting directory)
+@click.option(
+    "--report-dir",
+    "-o",
+    default="/tmp",
+    type=click.Path(exists=True, dir_okay=True, writable=True),
+    show_default=True,
+)
+@pass_config
+def validate_metrics(cfg: config.Validator, duration: datetime.timedelta, report_dir: str):
+    results_dir, tag = create_report_dir(report_dir)
+    res = TestResult(tag)
+    res.end_time = datetime.datetime.now(tz=datetime.UTC)
+    res.start_time = res.end_time - duration
+    click.secho("  * Generating build and node info ...", fg="green")
+    res.build_info, res.node_info = get_build_and_node_info(cfg.prometheus)
+    click.secho("  * Generating spec report ...", fg="green")
+    res.host_spec = get_host_spec()
+    script_result = ScriptResult(res.start_time, res.end_time)
+    res.validations = run_validations(cfg, script_result, results_dir)
+    click.secho("  * Generating validate metrics report file and dir", fg="green")
+    write_md_report(results_dir, res)
+
+    raise Exit(1) if not res.validations.passed else Exit(0)
```

Comment:
can't we just return 0 or 1 ?

Reply:
by default when a Click script is invoked as command line application (through BaseCommand.main()) the return value is ignored unless the standalone_mode is disabled in which case it’s bubbled through.
https://click.palletsprojects.com/en/stable/commands/#command-return-values